src/org/zaproxy/zap/extension/ascanrulesAlpha/SQLInjectionMsSQL.java

```diff
@@ -0,0 +1,319 @@
+				try {
+					sendAndReceive(msgAttack, false); //do not follow redirects
+					countTimeBasedRequests++;
+					}
+				catch (java.net.SocketTimeoutException e) {
+					//this is to be expected, if we start sending slow queries to the database.  ignore it in this case.. and just get the time.
+					if ( this.debugEnabled ) log.debug("The time check query timed out on ["+msgTimeBaseline.getRequestHeader().getMethod()+"] URL ["+msgTimeBaseline.getRequestHeader().getURI().getURI()+"] on field: ["+paramName+"]");
+				}
+				long modifiedTimeUsed = System.currentTimeMillis() - modifiedTimeStarted;
+
+				if ( this.debugEnabled ) log.debug ("Time Based SQL Injection test: ["+ newTimeBasedInjectionValue + "] on field: ["+paramName+"] with value ["+newTimeBasedInjectionValue+"] took "+ modifiedTimeUsed + "ms, where the original took "+ originalTimeUsed + "ms");
+				//add some small leeway on the time, since adding a 5 (by default) second delay in the SQL query will not cause the request
+				//to take a full 5 (by default) seconds longer to run than the original..
+				if (modifiedTimeUsed >= (originalTimeUsed + sleep*1000 - 200)) {
+					//takes more than 5(by default) extra seconds => likely time based SQL injection. Raise it
+					String extraInfo = Constant.messages.getString("ascanalpha.sqlinjection.mssql.alert.timebased.extrainfo", newTimeBasedInjectionValue, modifiedTimeUsed, paramValue, originalTimeUsed);
+
+					//raise the alert
+					bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, getName(), getDescription(),
+							getBaseMsg().getRequestHeader().getURI().getURI(), //url
+							paramName,  newTimeBasedInjectionValue,
+							extraInfo, getSolution(), msgAttack);
+
+					if (log.isDebugEnabled()) {
+						log.debug("A likely Time Based SQL Injection Vulnerability has been found with ["+msgAttack.getRequestHeader().getMethod()+"] URL ["+msgAttack.getRequestHeader().getURI().getURI()+"] on field: ["+paramName+"]");
+					}
+					return;
+				} //query took longer than the amount of time we attempted to delay it by
+			}  //for each time based SQL index
+			//end of check for time based SQL Injection
+		} catch (InvalidRedirectLocationException e) {
+			// Not an error, just means we probably attacked the redirect location
+		} catch (Exception e) {
+			//Do not try to internationalise this.. we need an error message in any event..
+			//if it's in English, it's still better than not having it at all.
+			log.error("An error occurred checking a url for MsSQL Injection vulnerabilities", e);
+		}
+	}
+
+	private String getSleepToken(int totalTimeInSeconds){
+		int hoursInTotalTime = totalTimeInSeconds / 3600;
+		totalTimeInSeconds %= 3600;
+		int minutesInTotalTime = totalTimeInSeconds / 60;
+		totalTimeInSeconds %= 60;
+		int secondsInTotalTime = totalTimeInSeconds;
+		String sleepToken = Integer.toString(hoursInTotalTime) + ":" + Integer.toString(minutesInTotalTime) + ":" + Integer.toString(secondsInTotalTime);
```

Comment:
- Could return the value immediately, we already know that's the sleep token (per methods' name).
 - It could use a StringBuilder.